src/GitHub.App/Controllers/UIController.cs

```diff
@@ -253,13 +258,21 @@ public void Start([AllowNull] IConnection connection)
                     .IsLoggedIn(hosts)
                     .Do(loggedin =>
                     {
-                        if (!loggedin && currentFlow != UIControllerFlow.Authentication)
+                        if (currentFlow != UIControllerFlow.Authentication)
                         {
-                            connectionAdded = (s, e) => {
-                                if (e.Action == NotifyCollectionChangedAction.Add)
-                                    uiProvider.AddService(typeof(IConnection), e.NewItems[0]);
-                            };
-                            connectionManager.Connections.CollectionChanged += connectionAdded;
+                            if (loggedin) // register the first available connection so the viewmodel can use it
+                                uiProvider.AddService(connectionManager.Connections.First(c => hosts.LookupHost(c.HostAddress).IsLoggedIn));
```

Comment:
It should never happen, but could there be some sort of race condition or something here? Should we use `FirstOrDefault` and check for null to be defensive?